internal/service/apikeyprojectassignment/data_source.go

```diff
@@ -0,0 +1,55 @@
+package apikeyprojectassignment
+
+import (
+	"context"
+
+	"github.com/hashicorp/terraform-plugin-framework/datasource"
+	"github.com/mongodb/terraform-provider-mongodbatlas/internal/common/conversion"
+	"github.com/mongodb/terraform-provider-mongodbatlas/internal/config"
+)
+
+var _ datasource.DataSource = &ds{}
+var _ datasource.DataSourceWithConfigure = &ds{}
+
+func DataSource() datasource.DataSource {
+	return &ds{
+		DSCommon: config.DSCommon{
+			DataSourceName: resourceName,
+		},
+	}
+}
+
+type ds struct {
+	config.DSCommon
+}
+
+func (d *ds) Schema(ctx context.Context, req datasource.SchemaRequest, resp *datasource.SchemaResponse) {
+	resp.Schema = conversion.DataSourceSchemaFromResource(ResourceSchema(ctx), &conversion.DataSourceSchemaRequest{
+		RequiredFields: []string{"project_id", "api_key_id"},
+	})
+}
+
+func (d *ds) Read(ctx context.Context, req datasource.ReadRequest, resp *datasource.ReadResponse) {
+	var tfModel TFModel
+	resp.Diagnostics.Append(req.Config.Get(ctx, &tfModel)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	connV2 := d.Client.AtlasV2
+	projectID := tfModel.ProjectId.ValueString()
+	apiKeys, err := ListAllProjectAPIKeys(ctx, connV2, projectID)
```

Comment:
we can leave a comment with CLOUDP ticket mentioning that ideally we have a single GET for this